### PR TITLE
Fix Open Point in time API

### PR DESCRIPTION
**Pull request by pquentin**
The remaining failures will be fixed when the point in time NEST tests will be removed.

### CODE REVIEW

File: specification/_global/open_point_in_time/OpenPointInTimeResponse.ts
Comment (l-trotta) [inline specification/_global/open_point_in_time/OpenPointInTimeResponse.ts:26]: _shards is [optional](https://github.com/elastic/elasticsearch/blob/5e455db10ecbb1a31cad58ecb1120a66fc50079f/server/src/main/java/org/elasticsearch/action/search/OpenPointInTimeResponse.java#L49)

Comment (l-trotta) [reply specification/_global/open_point_in_time/OpenPointInTimeResponse.ts:26]: we could merge this in main and wait for the backport?

Comment (pquentin) [reply specification/_global/open_point_in_time/OpenPointInTimeResponse.ts:26]: Right, we won't backport because 8.15 will never have `_shards`. (I could backport the `index_filter` part but I'm being lazy here.)

We agreed offline to wait for the `_shards` change to land in serverless before merging.

**Review by l-trotta — APPROVED** [review]

**Pull request merged**
